### DISCUSSION TITLE
Support for KeyCloud 15.0.2.

### DISCUSSION THREAD

**Discussion by onknows** [OpenIDC/mod_auth_openidc#726] [General]
Following provided instruction it seems mod_auth_openidc can no longer integrate with KeyCloud 15.0.2. KeyCloak responds with an error message which I think is unique in the sense I cannot find any information on it. When I navigate to my protected app it redirects to the vanity url https://myserver/myapp/redirect_uri?error=invalid_request&error_description=Missing+parameter%3A+code_challenge_method&state=fYELLNsjY08vx6hgbPEi6GfU6tA

```
Error:

OpenID Connect Provider error: invalid_request

Description:

Missing+parameter:+code_challenge_method
```

So it seems that KeyCloud expects mod_auth_openidc to send some parameter regarding code challenge.

As far as I can see mod_auth_openidc does provide a configuration option for that and also KeyCloud does not allow us to configure a default "code challenge".

This does not seem to be a known issue because there is KeyCloud or mod_auth_openidc documentation on "code challenges". Is this some new feature in KeyCloak that breaks mod_auth_openidc support?

What is the most recent KeyCloud version that mod_auth_openidc can integrate with?

**Comment by onknows** [comment]
The config that was missing is

`OIDCPKCEMethod S256`

**Reply by zandbelt** [reply]
can you confirm that this is happening for an Implicit flow i.e. OIDCResponseType != code?
